Add smoketest builds for openvpn3, openvpn3-linux and ovpn-dco

mattock opened the pull request.

mattock committed:
Add openvpn3-smoketest builds

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock committed:
Add openvpn3-linux smoketests

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock committed:
Enable ovpn-dco smoketest builds

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock commented:
The openvpn3-linux smoketest builds include both OpenSSL and MbedTLS builds. If those builds pass then the rest of the builds are launched.

The openvpn3 and ovpn-dco smoketests include just one build, followed by the normal builds if all goes well.

I've tested each of these individually, but independent testing would be good.

flichtenheld commented on buildbot-host/buildmaster/master.cfg:
```diff
@@ -604,6 +607,22 @@ factories.update(
 
+# openvpn3 smoketest
+factory = util.BuildFactory()
+factory = openvpn3AddCommonLinuxStepsToBuildFactory(factory, combo)
```
What is `combo` here supposed to mean. We are not in a `for` loop, are we? I know you only copied this, but still would like to clean this up.

mattock replied:
You're correct - copy and paste error. Not really sure what the implications are to be honest, but I'll clean it and the other places up.

flichtenheld commented on buildbot-host/buildmaster/master.cfg:
```diff
@@ -604,6 +607,22 @@ factories.update(
+
 # Basic openvpn3 builds
 factory = util.BuildFactory()
 factory = openvpn3AddCommonLinuxStepsToBuildFactory(factory, combo)
```
Copied from here

flichtenheld commented on buildbot-host/buildmaster/master.cfg:
```diff
@@ -656,6 +711,22 @@ factories.update(
 
+# ovpn-dco smoketest builds
+factory = util.BuildFactory()
+factory = ovpnDcoAddCommonLinuxStepsToBuildFactory(factory, combo)
```
`combo` again

flichtenheld commented on buildbot-host/buildmaster/master.cfg:
```diff
@@ -656,6 +711,22 @@ factories.update(
+
 # Basic ovpn-dco builds
 factory = util.BuildFactory()
 factory = ovpnDcoAddCommonLinuxStepsToBuildFactory(factory, combo)
```
Copied from here

dsommers commented on buildbot-host/buildmaster/master.cfg:
```diff
@@ -655,6 +674,24 @@ factories.update(
 )
 del factory
 
+# MbedTLS openvpn3-linux smoketest builds
```
I'm not sure this makes sense any more.  As of `v22_dev`, mbed TLS support has essentially not been accounted for.

mattock replied:
Are you saying we should disable mbed TLS build altogether or?

flichtenheld replied:
For openvpn3-linux and openvpn3 there is no one maintaining the mbedTLS port. So if there are any problems with it we just disable the offending build, unless it is trivial to fix. In the case of openvpn3-linux I think there are no other known mbedTLS builds left. For openvpn3 we still have some internally.

flichtenheld replied:
So yes, I think disabling mbedTLS builds for openvpn3-linux makes sense until someone actually offers to maintain that "port".

dsommers replied:
Yes, I'm pretty sure of dropping mbed TLS support in OpenVPN 3 Linux. It's not even configurable at the moment with the Meson setup in use now.

I'm going to remove the mention of mbed TLS in the BUILD.md as well for the `v23` release.

dsommers reviewed (changes requested):
I would not consider mbed TLS at the moment.

mattock committed:
Remove openvpn3-linux mbedtls builds

We don't support mbedtls builds anymore

URL: https://github.com/OpenVPN/openvpn-buildbot/pull/51
Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock committed:
Remove stray "combo" variable from openvpn3 and ovpn-dco builds

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock committed:
openvpn3-linux: remove useless config_opts parameter

With the meson build system we don't need any of the configure parameters

Signed-off-by: Samuli Seppänen <samuli.seppanen@gmail.com>

mattock commented:
I pushed three more commits that:

* Remove openvpn3-linux mbedtls builds (smoketest and normal)
* Remove unnecessary "combo" and "config_opts" parameters from openvpn3-linux, openvpn3 and ovpn-dco

I ran them through my private buildbot with one smoketest and one normal builder and all looks good afaics.
